src/components/PermissionsLoader.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+import useInventoryWritePermissions from '../hooks/useInventoryWritePermissions';
+import { ACTION_TYPES } from '../constants';
+
+const PermissionLoader = () => {
```

Comment:
Interesting way of using hook component to mutate redux!

Reply:
Yeah normally I would add this code to some top level component as a mount event, but I didn't want to duplicate the `usePermissions` hook API code.

And actually like this approach, as it's really just a 'component plugin' :smile: I am using it a lot in Sources UI.